dashboard/app/controllers/api/v1/teacher_feedbacks_controller.rb

```diff
@@ -0,0 +1,44 @@
+class Api::V1::TeacherFeedbacksController < Api::V1::JsonApiController
+  load_and_authorize_resource
+
+  # Use student_id, level_id, and teacher_id to lookup the feedback for a student on a particular level and provide the
+  # most recent feedback left by that teacher
+  def show_feedback_from_teacher
+    @feedback = TeacherFeedback.where(student_id: params[:student_id], level_id: params[:level_id], teacher_id: params[:teacher_id]).last
+    render json: @feedback
+  end
+
+  # Use student_id and level_id to lookup the most recent feedback from each teacher who has provided feedback to that
+  # student on that level
+  def show_feedback_for_level
+    query = <<-EOS
+      SELECT b.*
+      FROM (
+        SELECT teacher_id, max(created_at) as created_at
+        FROM teacher_feedbacks GROUP BY teacher_id) a
+      INNER JOIN teacher_feedbacks b ON a.teacher_id = b.teacher_id AND a.created_at = b.created_at
+      WHERE b.student_id = #{params[:student_id]} and b.level_id = #{params[:level_id]}
+    EOS
+
+    @level_feedbacks = TeacherFeedback.find_by_sql query
+
+    render json: {feedbacks: @level_feedbacks}
```

Comment:
Is this meant to be multiple feedbacks or a single one? If multiple, remove the `first` on my suggested query. If single, make the variable and key name singular as well so we're consistent.

Reply:
Note the data schemas for these 2 APIs are inconsistent. The first serializes the `TeacherFeedback` directly, whereas this one wraps it in a `feedbacks` container. Ideally these would be consistent.